File: src/python/config/read_config.py
import pandas as pd
import yaml

from src.python.path import pathOfDataFile


# read CSV data file
def readCSVDataFile(fname: str):
    fpath = pathOfDataFile(fname)
    return pd.read_csv(fpath)


# read YAML config file
def readYAMLDataFile(fname: str):
    fpath = pathOfDataFile(f"{fname}.yml")
    fhandle = open(fpath, 'r', encoding='utf-8')
    ret = yaml.load(stream=fhandle, Loader=yaml.FullLoader)
    fhandle.close()
    return ret

File: src/R/config/read_config.R
library(yaml)

source("src/R/path.R")


# read CSV data file
readCSVDataFile <- function (fname) {
    fpath <- pathOfDataFile(fname)
    return(read.csv(
        fpath,
        sep=',',
        quote='"',
        encoding='utf-8'
    ))
}


# read YAML config file
readYAMLDataFile <- function (fname) {
    fpath <- pathOfDataFile(paste0(fname, ".yml"))
    fhandle <- file(fpath, "r")
    ret <- read_yaml(file=fhandle, fileEncoding="utf-8")
    close(fhandle)
    return(ret)
}
